metadata-models/src/main/pegasus/com/linkedin/join/JoinProperties.pdl

```diff
@@ -0,0 +1,79 @@
+namespace com.linkedin.join
+
+import com.linkedin.common.TimeStamp
+import com.linkedin.common.DatasetUrn
+import com.linkedin.common.CustomProperties
+
+/**
+ * Properties associated with a Join
+ */
+@Aspect = {
+  "name": "joinProperties"
+}
+record JoinProperties includes CustomProperties {
+
+  /**
+   * Name of the Join
+   */
+  @Searchable = {
+    "fieldType": "TEXT_PARTIAL",
+    "enableAutocomplete": true,
+    "boostScore": 10.0
+  }
+  name: string
+
+  /**
+  * First dataset in the join (no directionality)
+  */
+   @Relationship = {
+     "name": "joinA",
+     "entityTypes": [ "dataset" ]
+   }
+  @Searchable = {
+    "fieldType": "TEXT_PARTIAL",
+    "enableAutocomplete": true,
+    "boostScore": 10.0
+  }
+  datasetA: DatasetUrn
+
+  /**
+  * Second dataset in the join (no directionality)
+  */
+   @Relationship = {
+     "name": "joinB",
+     "entityTypes": [ "dataset" ]
+   }
+  @Searchable = {
+    "fieldType": "TEXT_PARTIAL",
+    "enableAutocomplete": true,
+    "boostScore": 10.0
+  }
+  datasetB: DatasetUrn
+
+  /**
+   * JoinFieldMapping (in future we can make it an array)
+   */
+  joinFieldMappings: JoinFieldMapping
+
+  /**
+   * A timestamp documenting when the asset was created in the source Data Platform (not on DataHub)
+   */
+  @Searchable = {
+    "/time": {
+      "fieldName": "createdAt",
+      "fieldType": "DATETIME"
+    }
+  }
+  created: optional TimeStamp
+
+  /**
+   * A timestamp documenting when the asset was last modified in the source Data Platform (not on DataHub)
+   */
+  @Searchable = {
+    "/time": {
+      "fieldName": "lastModifiedAt",
+      "fieldType": "DATETIME"
+    }
+  }
+  lastModified: optional TimeStamp
```

Comment:
how about using `AuditStamp` instead of `TimeStamp`? it just gives us some more optional fields to be more flexible in the future and requires the `actor` field to be populated

Reply:
Addressed.